lib/fizzy/instantiate.cpp

```diff
@@ -174,6 +174,13 @@ std::tuple<bytes_ptr, Limits> allocate_memory(const std::vector<Memory>& module_
     static const auto bytes_delete = [](bytes* b) noexcept { delete b; };
     static const auto null_delete = [](bytes*) noexcept {};
 
+    if (memory_pages_limit > MemoryPagesMaxLimit)
+    {
+        throw instantiate_error{"hard memory limit cannot exceed " +
```

Comment:
Nit: I would mention "during instantiate" or something like that in the commit log for this change.